### DISCUSSION TITLE
Entity vs Resource

### DISCUSSION THREAD

**Discussion by dbones** [dotnet/dotnet-operator-sdk#634] [Q&A]
In the latest code it looks like the `resource` terminology is being replaced fully with `entity`

`IResourceController<TEntity>` -> `IEntityConrtroller<TEntity>`

I was wondering if I have missed my understanding in Kubernetes, they seem to call everything resources (for example Custom Resources Definitions, CRD)

Should we not use the same terminology?

i.e. `IResourceController<TResourceDefinition>`  (i little verbose, but hopefully it makes sense)

**Comment by buehler** [comment]
actually.... YES :D
I tried to rename everything to use c# terminology (most of the time you find some "models" or "entities" or whatnot.) But to use "resource" would be better.

I think I'll rename everything to match kubernetes terminology. In v7 and below, there was a mix which bothered me for a long time.

**Reply by buehler** [reply]
However, the thing is, that Kubernetes itself uses the `KubernetesEntityAttribute`... so it is kind of mixed up.

**Reply by dbones** [reply]
I feel the pain

On this page https://kubernetes.io/docs/reference/glossary/?fundamental=true

Everything here but the kubernetes Object is a Resource, the Object is referred to as an Entity

it may be worth adding a discussion with the K8s team to see if we can get them to change and be more consistent. Which in turn helps here :)

(I know that using Resource would help me train others where i work)

**Reply by buehler** [reply]
Well then, let's settle it and go for "resource" barring the "KubernetesEntity" attribute which is kind of wrong named then. after the change, we can bring the discussion over to the kubernetes client team.

**Reply by dbones** [reply]
LGTM :D

**Reply by buehler** [reply]
f*ck. totally forgot to rename everything 🙈 
i guess it stays as it is for now.
